test/e2e/deployment.go

```diff
@@ -202,16 +211,38 @@ func stopDeployment(c *clientset.Clientset, oldC client.Interface, ns, deploymen
 	options := api.ListOptions{LabelSelector: selector}
 	rss, err := c.Extensions().ReplicaSets(ns).List(options)
 	Expect(err).NotTo(HaveOccurred())
-	Expect(rss.Items).Should(HaveLen(0))
+	// RSes may be created by overlapping deployments right after this deployment is deleted, ignore them
+	if len(overlapWith) == 0 {
+		Expect(rss.Items).Should(HaveLen(0))
+	} else {
+		ignoreOverlapRSes := []extensions.ReplicaSet{}
```

Comment:
We need a better name for this since these replica sets are not to be ignored. Same comment for pods below.